### ISSUE TITLE
Missing columns bug

### ISSUE THREAD

**Issue by fegu** [qrilka/xlsx#23]
In some cases there are missing data in the parsed Excel sheet. To reproduce:

Get [small example sheet](http://applikativ.no/TNM.xlsx)

in ghci:

import Codec.Xlsx
import qualified Data.ByteString.Lazy as L
import qualified Data.Map as Map

e <- L.readFile "TNM.xlsx"
let lx = _wsCells $ snd $ head $ Map.toList $ _xlSheets $ toXlsx e
map (\c -> (Map.!) lx (2,c)) [2..6] -- this is ok
map (\c -> (Map.!) lx (5,c)) [2..6] -- column is missing

Compare the result of the last command to the actual Excel file. Column E is missing and all other data shifted left.

**Comment by qrilka** [comment]
@fegu could you show your GHCi output?
Mine shows 5 columns for both rows:
```
λ> map (\c -> (Map.!) lx (2,c)) [2..6] -- this is ok
[Cell {_cellStyle = Just 2, _cellValue = Just (CellDouble 42254.0)},Cell {_cellStyle = Nothing, _cellValue = Just (CellText "5504542218")},Cell {_cellStyle = Nothing, _cellValue = Just (CellText "SYSTOR VEST AS")},Cell {_cellStyle = Nothing, _cellValue = Just (CellText "5504542218")},Cell {_cellStyle = Nothing, _cellValue = Just (CellText "SYSTOR VEST AS")}]
λ> map (\c -> (Map.!) lx (5,c)) [2..6] -- column is missing
[Cell {_cellStyle = Just 2, _cellValue = Just (CellDouble 42254.0)},Cell {_cellStyle = Nothing, _cellValue = Just (CellText "5504542218")},Cell {_cellStyle = Nothing, _cellValue = Just (CellText "SYSTOR VEST AS")},Cell {_cellStyle = Nothing, _cellValue = Just (CellText "41109511")},Cell {_cellStyle = Nothing, _cellValue = Just (CellText "EIRIK MAGNUSSEN")}]
```

**Comment by fegu** [comment]
Hi, yes, there are 5 columns, but the content I get is wrong. One column is missing and all other data for that row shifted left to fill the void.

    > map (\c -> (Map.!) lx (2,c)) [2..6] -- this is ok
    [Cell {_cellStyle = Just 2, _cellValue = Just (CellDouble 42254.0)},Cell {_cellStyle = Nothing, _cellValue = Just (CellText "5504542218")},Cell {_cellStyle = Nothing, _cellValue = Just (CellText "SYSTOR VEST AS")},Cell {_cellStyle = Nothing, _cellValue = Just (CellText "5504542218")},Cell {_cellStyle = Nothing, _cellValue = Just (CellText "SYSTOR VEST AS")}]

    > map (\c -> (Map.!) lx (5,c)) [2..6]
    [Cell {_cellStyle = Just 2, _cellValue = Just (CellDouble 42254.0)},Cell {_cellStyle = Nothing, _cellValue = Just (CellText "5504542218")},Cell {_cellStyle = Nothing, _cellValue = Just (CellText "SYSTOR VEST AS")},Cell {_cellStyle = Nothing, _cellValue = Just (CellText "EIRIK MAGNUSSEN")},Cell {_cellStyle = Nothing, _cellValue = Just (CellText "Faste priser")}]

Since you get correct results, while I don't, I thought about comparing versions. I am using GHC7.8.4 and xlsx-0.1.1 on Windows. However, it seems this bug is fixed in xlsx-0.1.1.1 as I get the correct (your) result now as well. Before you close the issue, perhaps look into why?

**Comment by qrilka** [comment]
Thanks @fegu
It appears that there was a bug with empty shared strings, i.e. your file contains the following entry:
```xml
<si><t/></si>
```
And as rich string support required some refactoring of shared strings parsing code this bug was also fixed as a side effect.
